### DISCUSSION TITLE
Move Video Capture Device on Linux

### DISCUSSION THREAD

**Discussion by royveshovda** [exeldro/obs-move-transition#145] [Q&A] [answered]
Hi. Is the "Move Video Capture Device" filter suppose to work on Linux?
If yes, I need some help getting it to show up.
If no, I am happy to help debug and get this working, if that can be of any help.

A bit of background:
I have the Insta360 Link camera. I can dual-boot my computer. And on Windows everything shows up as expected, and I can move the pan/tilt/zoom on the camera, as expected. On Linux (Pop!_OS, the filter "Filter Video Capture Device does not show up. Only "Move Action/Source/Value"

**Comment by exeldro** [comment, marked as answer]
The "Move Video Capture Device" filter uses directshow to control camera movement which is only available on windows. I have no idea how this would work on Linux.

**Reply by pcpepik** [reply]
Should be easy to implement, you can use official linux-v4l2 plugin as a reference: https://github.com/obsproject/obs-studio/tree/master/plugins/linux-v4l2

PS: You can think of V4L2 as a linux equivalent to directshow on windows: https://en.wikipedia.org/wiki/Video4Linux

<details>
<summary>Show screenshot</summary>

![image](https://github.com/exeldro/obs-move-transition/assets/2814558/13c48581-ca7d-4775-a745-190d8d513ea2)

</details>

**Comment by royveshovda** [comment]
Fair enough. I assume Linux would be possible using UCV (similar to  GUCVView: https://guvcview.sourceforge.net/ ). I can control pan/til/zoom using that app. Probably a bit involved to get up and running, though. I am happy to assist if you want to pursue this path. Understandable if you choose not to.

**Comment by shanness** [comment]
Would this help?  It seems to explicitly work for linux.
https://github.com/glikely/obs-ptz/
